projects/distributed-tracing/src/shared/graphql/request/handlers/traces/export-spans-graphql-query-handler.service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@angular/core';
+import { TimeDuration, TimeRangeService, TimeUnit } from '@hypertrace/common';
+import { GraphQlHandlerType, GraphQlQueryHandler, GraphQlSelection } from '@hypertrace/graphql-client';
+import { GlobalGraphQlFilterService } from '../../../model/schema/filter/global-graphql-filter.service';
+import { GraphQlFilter } from '../../../model/schema/filter/graphql-filter';
+import { GraphQlIdFilter } from '../../../model/schema/filter/id/graphql-id-filter';
+import { GraphQlTimeRange } from '../../../model/schema/timerange/graphql-time-range';
+import { resolveTraceType, TraceType } from '../../../model/schema/trace';
+import { GraphQlArgumentBuilder } from '../../builders/argument/graphql-argument-builder';
+
+@Injectable({ providedIn: 'root' })
+export class ExportSpansGraphQlQueryHandlerService
+  implements GraphQlQueryHandler<GraphQlExportSpansRequest, string | undefined> {
+  public readonly type: GraphQlHandlerType.Query = GraphQlHandlerType.Query;
+  private readonly argBuilder: GraphQlArgumentBuilder = new GraphQlArgumentBuilder();
+
+  public constructor(
+    private readonly timeRangeService: TimeRangeService,
+    private readonly globalGraphQlFilterService: GlobalGraphQlFilterService
+  ) {}
+
+  public matchesRequest(request: unknown): request is GraphQlExportSpansRequest {
+    return (
+      typeof request === 'object' &&
+      request !== null &&
+      (request as Partial<GraphQlExportSpansRequest>).requestType === EXPORT_SPANS_GQL_REQUEST
+    );
+  }
+
+  public convertRequest(request: GraphQlExportSpansRequest): GraphQlSelection {
+    const timeRange = this.buildTimeRange(request.timestamp);
+
+    return {
+      path: 'exportSpans',
+      arguments: [
+        this.argBuilder.forLimit(request.limit),
+        this.argBuilder.forTimeRange(timeRange),
+        ...this.argBuilder.forFilters(
+          this.globalGraphQlFilterService.mergeGlobalFilters(resolveTraceType(request.traceType), [
+            this.buildTraceIdFilter(request)
+          ])
+        )
+      ],
+      children: [
+        {
+          path: 'result'
+        }
+      ]
+    };
+  }
+
+  public convertResponse(response: ExportSpansResponse): string | undefined {
+    return response.result;
+  }
+
+  private buildTraceIdFilter(request: GraphQlExportSpansRequest): GraphQlFilter {
+    return new GraphQlIdFilter(request.traceId, resolveTraceType(request.traceType));
+  }
+
+  protected buildTimeRange(timestamp?: Date): GraphQlTimeRange {
+    let timeRange;
+    if (timestamp) {
+      const duration = new TimeDuration(30, TimeUnit.Minute);
+      timeRange = {
```

Comment:
nit: return directly rather than assigning to a mutable var (also, there's a static constructor that makes it a bit cleaner):
`return GraphQlTimeRange.fromTimeRange(this.timeRangeService.getCurrentTimeRange());`

Reply:
Done!!